src/modules/developer-guide/pages/server-plugins/index.adoc

```diff
@@ -0,0 +1,145 @@
+include::ROOT:partial$variables.adoc[]
+
+= Server plugins
+
+{PRODUCT_NAME} server can be extended using https://pkg.go.dev/plugin[Go plugins].
+You can modify default behaviour of the server and provide new functionalities like workflow functions and types.
+
+== Basic plugin structure
+
+[source,go]
+----
+package main
+
+type (
+	examplePlugin struct {} <1>
+)
+
+func CortezaPlugin() interface{} { <2>
+	return &examplePlugin{}
+}
+----
+. Struct that holds
+. Be sure you return an `interface{}`, otherwise {PRODUCT_NAME} can not cast it to plugin
+
+== Supported hooks
```

Comment:
[Personal opinion; would like to see it implemented, but can be ignored]

For smaller bits of information, I would prefer they get grouped under a single sub-title such as `Supported hooks` in your case.
I would then list all of the different hooks in a table (see the one I've quickly thrown together).
Have a scroll through http://localhost:8080/corteza-docs/2021.9/integrator-guide/automation/workflows/index.html this pattern is used all over the place.
If a thing would grow out of control (with a lot of images/sub-sub-sections), then this can be moved out of the table.

Such an approach makes it a bit more compact and it doesn't clutter the TOC for no added benefit.

Reply:
See the bellow comments that explain how a table row should be structured and what each bit represents